typing_extensions/src_py3/test_typing_extensions.py

```diff
@@ -1458,6 +1460,183 @@ def test_total(self):
+    from typing_extensions import Annotated, get_type_hints
+
+    class AnnotatedTests(BaseTestCase):
+
+        def test_repr(self):
+            self.assertEqual(
+                repr(Annotated[int, 4, 5]),
+                "typing_extensions.Annotated[int, 4, 5]"
+            )
+
+        def test_flatten(self):
+            A = Annotated[Annotated[int, 4], 5]
+            self.assertEqual(A, Annotated[int, 4, 5])
+            self.assertEqual(A.__metadata__, (4, 5))
+            self.assertEqual(A.__origin__, int)
+
+        def test_hash_eq(self):
+            self.assertEqual(len({Annotated[int, 4, 5], Annotated[int, 4, 5]}), 1)
+            self.assertNotEqual(Annotated[int, 4, 5], Annotated[int, 5, 4])
+            self.assertNotEqual(Annotated[int, 4, 5], Annotated[str, 4, 5])
+            self.assertNotEqual(Annotated[int, 4], Annotated[int, 4, 4])
+            self.assertEqual(
+                {Annotated[int, 4, 5], Annotated[int, 4, 5], Annotated[T, 4, 5]},
+                {Annotated[int, 4, 5], Annotated[T, 4, 5]}
+            )
+
+        def test_instantiate(self):
+            class C:
+                classvar = 4
+
+                def __init__(self, x):
+                    self.x = x
+
+                def __eq__(self, other):
+                    if not isinstance(other, C):
+                        return NotImplemented
+                    return other.x == self.x
+
+            A = Annotated[C, "a decoration"]
+            a = A(5)
+            c = C(5)
+            self.assertEqual(a, c)
+            self.assertEqual(a.x, c.x)
+            self.assertEqual(A.classvar, C.classvar)
+
+            MyCount = Annotated[typing_extensions.Counter[T], "my decoration"]
+            self.assertEqual(MyCount([4, 4, 5]), {4: 2, 5: 1})
+            self.assertEqual(MyCount[int]([4, 4, 5]), {4: 2, 5: 1})
+
+
+        def test_cannot_subclass(self):
+            with self.assertRaises(TypeError):
+                class C(Annotated):
+                    pass
+
+        def test_pickle(self):
+            samples = [typing.Any, typing.Union[int, str],
+                       typing.Optional[str], Tuple[int, ...],
+                       typing.Callable[[str], bytes]]
+
+            for t in samples:
+                x = Annotated[t, "a"]
+
+                for prot in range(pickle.HIGHEST_PROTOCOL + 1):
+                    with self.subTest(protocol=prot, type=t):
+                        pickled = pickle.dumps(x, prot)
+                        restored = pickle.loads(pickled)
+                        self.assertEqual(x, restored)
+
+            global _Annotated_test_G
+
+            class _Annotated_test_G(Generic[T]):
+                x = 1
+
+            G = Annotated[_Annotated_test_G[int], "A decoration"]
+
+            c = G()
+            c.foo = 42
+            c.bar = 'abc'
+            for proto in range(pickle.HIGHEST_PROTOCOL + 1):
+                z = pickle.dumps(c, proto)
+                x = pickle.loads(z)
```

Comment:
Here you are testing pickling of a normal instance, instead you should (un-)pickle `G` itself.